src/library.js

```diff
@@ -862,10 +862,83 @@ const createPilingJs = (rootElement, initOptions = {}) => {
           .then(textures => textures.map(createPreview))
       : Promise.resolve([]);
 
-    return Promise.all([renderImages, renderPreviews]).then(
+    return [renderImages, renderPreviews];
+  };
+
+  const updateItemTexture = (updatedItems = []) => {
+    const {
+      items,
+      pileItemAlignment,
+      pileItemRotation,
+      previewSpacing
+    } = store.getState();
+
+    // eslint-disable-next-line no-param-reassign
+    updatedItems = updatedItems.length ? updatedItems : [...items];
+
+    Promise.all(createImagesAndPreviews(updatedItems)).then(
+      ([renderedImages, renderedPreviews]) => {
+        const { piles } = store.getState();
+        renderedImages.forEach((image, index) => {
+          const id = updatedItems[index].id || index;
+          const preview = renderedPreviews[index];
+          if (renderedItems.has(id)) {
+            renderedItems.get(id).replaceImage(image, preview);
+          }
+        });
+        updatedItems.forEach((item, index) => {
+          const id = item.id || index;
+          if (pileInstances.has(id)) {
+            const pile = pileInstances.get(id);
+            const pileState = piles[id];
+            pile.replaceImage();
```

Comment:
What is going on here? 😂 A pile doesn't have any image. It should only know about items. And why is there no argument? Is this some kind of `itemImageHasUpdated()` function?

Reply:
Oh it's actually replacing the pile-items images. I'm sorry the name is unclear. Maybe I should call it `replaceItemsImage()`? 

In `pile.js` it's taking `pileItem.item.image` to replace `pileItem.image` so there is no argument 😂